src/routes/labs-route.js

```diff
@@ -0,0 +1,70 @@
+const config = require('config');
+const multer = require('multer');
+const GridFsStorage = require('multer-gridfs-storage');
+const path = require('path');
+const crypto = require('crypto');
+const express = require('express');
+
+const storage = new GridFsStorage({
+  url: config.mongo,
+  connectionOpts: { useNewUrlParser: true },
+  file: (req, file) => {
+    return new Promise((resolve, reject) => {
+      crypto.randomBytes(16, (err, buf) => {
+        if (err) {
+          return reject(err);
+        }
+        const fileName = buf.toString('hex') + path.extname(file.originalname);
+
+        const fileInfo = {
+          filename: fileName,
+          bucket: 'uploads',
+        };
+        return resolve(fileInfo);
+      });
+    });
+  },
+});
+
+const upload = multer({ storage });
+
+module.exports = function createLabsRoute(repository) {
+  const router = express.Router();
+
+  router.route('/')
+    .get(list)
+    .post(upload.single('lab'), add);
+
+  router.route('/:filename')
+    .get(view);
+
+  router.route('/:id')
+    .delete(remove);
+
+  return router;
+
+  function add(req, res) {
+    if (!req.file) {
+      res.status(400).json({ err: 'File not recived ' });
+    } else res.json({ succes: 'File added' });
+  }
+
+  async function list(req, res) {
+    const resultFiles = await repository.list();
+    if (!resultFiles || resultFiles.length === 0) {
```

Comment:
why do you need 2 checks?

Reply:
If resultFiles is undefined it will be considered as false.